python_modules/dagster/dagster/core/definitions/solid_definition.py

```diff
@@ -213,6 +213,19 @@ def is_graph_job_op_node(self) -> bool:
     def compute_fn(self) -> Union[Callable[..., Any], "DecoratedSolidFunction"]:
         return self._compute_fn
 
+    def is_from_decorator(self) -> bool:
+        from .decorators.solid_decorator import DecoratedSolidFunction
+
+        return isinstance(self._compute_fn, DecoratedSolidFunction)
+
+    def get_output_annotation(self) -> Any:
+        if not self.is_from_decorator():
+            raise DagsterInvalidInvocationError(
+                f"Attempted to get output annotation for {self.describe_op()}, "
+                "which was not constructed from a decorated function."
+            )
+        return self.compute_fn.get_output_annotation()
```

Comment:
found myself often repeating the pattern of cast(DecoratedSolidFunction, solid_def.compute_fn). This is just a clearer way of doing that, and can probably be used in a bunch more places (grep `cast(DecoratedSolidFunction` in the codebase and you'll see what I mean)